### DISCUSSION TITLE
Support Azure DevOps Git authentication via azure identity library

### DISCUSSION THREAD

**Discussion by dsteeley** [renovatebot/renovate#30541] [Suggest an Idea]
### Azure Devops auth via azure identity

It's possible to use the azure identity library to generate Bearer tokens that can be used for Azure DevOps git operations. 
Would you be open to a PR expanding Renovate's Azure DevOps support to include authenticating with the azure_identity library. 

This would support self-hosted scenarios where the environment is already authenticated removing the need for passing long lived secrets to the pipeline.

**Comment by rarkins** [comment]
How long are such tokens valid for?

**Reply by dsteeley** [reply]
I believe [60-90 minutes](https://learn.microsoft.com/en-us/entra/identity-platform/access-tokens#token-lifetime). I suggest an implementation where a new token is generated for each repository to mitigate the risk of expiry.

**Comment by rarkins** [comment]
See https://github.com/renovatebot/renovate/discussions/30930 for a generic approach to this.

An alternative would be to support commonly used libraries "first class" by adding awareness of them to Renovate, such as hostType=azure-identity

**Reply by dsteeley** [reply]
I'm keen to help get this working so will reply on your linked discussion. 

I agree that the generic approach will result in this being supportable rather than special casing the azure host type. It may be required to add your alternative suggestion of hostType=azure-identity when supporting bearer token authentication. Package versions can be queried via the `azure` host type which I'm not sure would be authenticated through the generic mechanism listed above?
